### ISSUE TITLE
Problem with plsr function

### ISSUE THREAD

**Issue by Zhuxitong** [bhmevik/pls#18]
Hi,

When I run following command, all goes fine.
```
> plsr <- pls::plsr(cls ~ datmat, method = "oscorespls", ncomp = 7)
```

However when I add validation parameter, I met below problem:
```
> plsr <- pls::plsr(cls ~ datmat, method = "oscorespls", ncomp = 7,  validation = "CV")
Error in cvsegments(nobj, k = segments, type = segment.type) : 
  More segments than observations requested
```

What's wrong with this? Is this a bug?

**Comment by khliland** [comment]
Hi Zhuxitong,

When you specify `validation = "CV"` without any more parameters, the default is to use 10 cross-validation segments. Could it be that your data has fewer than 10 objects? The code works as it should on my computer using a larger data set.

Best regards,
Kristian

**Comment by Zhuxitong** [comment]
Hi khliland,

Thanks for your timely response. I also detect this problem because I tried following code:
```
pls::plsr(cls ~ datmat, method = "oscorespls", ncomp = 5, validation = "CV")
&
pls::plsr(cls ~ datmat, method = "oscorespls", ncomp = 5, validation = "CV", segments = 5)
```

The second works fine. Indeed in my data, I have only 8 samples so clearly it's impossible to perform 10 cross-validation. 

However in the *PLSDA.CV* function,  only four parameters are supported( mSetObj, methodName, compNum and choice). So how can I set the segments number parameter here?
